File: archive/cpp/2021/nov/sum-of-left-leaves.cpp
#include "common.h"

class Solution
{
public:
    int sumOfLeftLeaves(TreeNode *root)
    {
        if (!root)
        {
            return 0;
        }
        int l = 0, r = 0;
        if (root->left)
        {
            if (!root->left->left && !root->left->right)
            {
                l = root->left->val;
            }
            else
            {
                l = sumOfLeftLeaves(root->left);
            }
        }
        if (root->right)
        {
            r = sumOfLeftLeaves(root->right);
        }
        return l + r;
    }
};

File: archive/go/2020/aug/sum-of-left-leaves.go
package aug

import "leetcode/common"

func sum(root *common.TreeNode) int {
	return helper2(root, false)

}

func helper2(root *common.TreeNode, isLeft bool) int {
	if root == nil {
		return 0
	}
	if root.Left != nil && root.Right != nil {
		if isLeft {
			return root.Val
		} else {
			return 0
		}
	}
	return helper2(root.Left, true) + helper2(root.Right, false)
}
